consensus/event_loop.go

```diff
@@ -71,8 +72,12 @@ func (fc *fastHotStuffEventLoop) Init(
 		return errors.Wrap(err, "FastHotStuffEventLoop.Init: ")
 	}
 
-	// We track the current view here so we know which view to time out on later on.
 	fc.currentView = tip.Block.GetView() + 1
+	// If we have a checkpoint latest view from the blockchain that is greater than our tip's view,
+	// we update the current view to the checkpoint's latest view.
+	if checkpointLatestView > fc.currentView {
+		fc.currentView = checkpointLatestView
+	}
```

Comment:
You can also move this up to line 66 so the validation happens before the tip and safe blocks are stored
```suggestion
	// The currentView must be higher than the tip block's current view
	if currentView < tip.GetView() + 1 {
		return errors.new("FastHotStuffEventLoop.Init: currentView is lower than the tip block's view")
	}
	fc.currentView = currentView
```